### ISSUE TITLE
new data entry form change requests

### ISSUE THREAD

**Issue by DerekSikes** [ArctosDB/arctos#6551]
Please return on the data entry form the sort sequence that we've been using forever - record event sorted above collection event, sorted above locality so locality is entered last but before parts.

The 'pull specific locality from verbatim' button is particularly useless if everyone is entering the specific locality before the verbatim locality now- there's nothing to pull. But there should be. It is FAR better to enter verbatim first then make any needed corrections to the specific locality later, not the reverse.

And seems all my prior data entry profiles are gone & need to be remade using this new form? Perhaps I can resort these myself? But even so, I feel the default should be as specified above.

**Priority**
Please assign a priority-label. Unprioritized issues get sent into a black hole of despair.
high

**Comment by DerekSikes** [comment]
And I thought we were going to get event attributes 'collection method' and 'habitat'?

**Comment by DerekSikes** [comment]
So I've set a customized data entry profile to show only 1 identifier because this will handle 95% of cases - if during data entry I need to enter a 2nd identifier, how to get the form to provide me the needed fields? Can there be a button 'add more' or something like a standard UI? Hopefully I won't have to go create another data entry profile for every such 'need one more' for all these attributes, etc.

**Comment by genevieve-anderegg** [comment]
I agree. 
Can we also please make the event_verbatim_locality field bigger and/or "stretchable"? It has become very small and now you have to use the arrow keys to scroll through and read the whole field. A lot of our verbatim localities are much longer than the size of this field.
![image](https://github.com/ArctosDB/arctos/assets/83782023/9704b7e7-d5cf-4d17-b425-c9f4cb147979)

**Comment by Jegelewicz** [comment]
@dustymc please advise if any of these need to be split out into separate issues or how you would like this to be handled.

**Comment by Jegelewicz** [comment]
Also, @DerekSikes @genevieve-anderegg how is any of this a bug? What is preventing you from getting work done? I get the issue described, but does it mean that Arctos is non-functional or just difficult to use?

**Comment by genevieve-anderegg** [comment]
That was an accident, it just makes Arctos difficult to use

**Comment by genevieve-anderegg** [comment]
You and I added labels at the same time, haha

**Comment by dustymc** [comment]
>how you would like this to be handled

I think this is all non-technical, so I'm up for anything. There was a several-months very publicized process which lead here, I think we can't make arbitrary changes without going through some sort of equally-open process. Perhaps a good task for an ad hoc committee? (And we really need to find some way to make decisions and stick with them!)

Perhaps a temporary milestone would be useful for organization - there are already several Issues, they should probably all be dealt with as a package.

The big thing that we've agreed not to change is the structure of the table, and I'm not sure having it conflict with the order of the form is great UI, but it is "just UI" and I can put whatever wherever.

**Comment by Jegelewicz** [comment]
I think that these things are sorta no-brainers:

1. Order of data entry should be record event sorted above collection event, sorted above locality (because "The 'pull specific locality from verbatim' button is particularly useless if everyone is entering the specific locality before the verbatim locality now").
2. make the event_verbatim_locality field "stretchable"

Both of those are "just UI" and I don't think anyone will complain about either of them. We can discuss briefly in AWG on Thursday if needed. @ArctosDB/arctos-working-group-officers

**Comment by dustymc** [comment]
>no-brainers

(1) - There were at least two very long meetings focused entirely on this, and they resulted in what you see. I'm sure I pointed out that this is different than the order has always been. It seems that SOMEONE (collectively, maybe) has some sort of an opinion on this! (My only ask is that we find something The Community can live with and stop making arbitrary changes. A 9 month process should have had this all sorted out?!)

(2) Yea, that doesn't seem like it could be or become controversial, but I'd still rather get it as part of a bigger process.

**Comment by campmlc** [comment]
I definitely support change the event locality stack order to revert to the previous order. No brainer there. I also strongly support option of "add more" for attributes, parts, etc so if you have one record with an extra value you don't have to re- edit the form, which is very painful and too much to ask if students. 
I have numerous other issues related to the Pull function I will file separately.

**Comment by campmlc** [comment]
9 months worth of discussion without ability to preview what users actually see and experience is not to be equated with approval of a final product. Given we had no sandbox to preview, the issues we are dealing with are minor so far, and the tweaks as well. But we should expect them  under these circumstances. There is a big difference between looking at endless GitHub issues and Google sheets and working with an actual UI.

**Comment by Jegelewicz** [comment]
> The big thing that we've agreed not to change is the structure of the table, and I'm not sure having it conflict with the order of the form is great UI, but it is "just UI" and I can put whatever wherever.

I had no idea that the order of the bulkloader table was going to set the order of data entry. Although that makes sense when I see it now, I really did not know that is what we were doing! And realistically, there is no reason for them to be in the same order is there?

**Comment by DerekSikes** [comment]
We recently had the ability to create a customized data entry form that could be selected by anyone working in that collection. Is that gone? are the customized forms only tied to the one user who customized them? I'd like to make a customized form that all my lab techs & students etc can use, with preselected values etc to reduce their error rates.

**Comment by Jegelewicz** [comment]
> ability to create a customized data entry form that could be selected by anyone working in that collection.

Hit the customize at the bottom and see this

![image](https://github.com/ArctosDB/arctos/assets/5725767/2fe4c0f1-f641-4a58-8ebf-dec6bea8ba63)

Go to Profiles Home to see the saved profiles available.

**Comment by DerekSikes** [comment]
so before, when one clicked data entry one saw the profiles... now one has to click data entry, go to the bottom and click customize and the click profiles home to see the profiles? Why not just have it like it was - the first screen after clicking data entry should be the profiles screen?

**Comment by campmlc** [comment]
Agree with @DerekSikes. Why make random changes that require figuring out additional clicks and cause  confusion? Consistency between upgrades to the extent possible makes everyone's life easier, and prevents frustration down a very long line of impacted users.

**Comment by AdrienneRaniszewski** [comment]
agree with everything folks above have noted about the new data entry screen and profiles. (fewer clicks to get to profiles, verbatim before specific, larger locality field, an add more button) Have a couple observations of my own...

I can't figure out how to rearrange the data blocks. I used to be able to arrange them so I could see everything I needed without scrolling. and to match the order of our data sheets. a big time saver!

Also liked the way we used to be able to customize one block at a time. I can’t figure that out either and the huge list on the customize screen is a bit overwhelming. I can live with this the way it it, but man, the other way was so simple! Like that I can edit an existing profile though. a very helpful improvement!

might have a few more comments once I re-create all my profiles and start entering data again.

definitely appreciate the more consistent field names and the expanded parts and attributes sections! Thanks!

**Comment by AdrienneRaniszewski** [comment]
missing the function to fill began and ended date from the verbatim date

**Comment by mkoo** [comment]
HI all, thanks everyone for the feedback: now that we have the new form up and some discussion time at the AWG today, I'm taking some time to summarize the requests (add if I'm missing yours or you have a new one):

- [ ] allow customizable ordering of sections. This would address the Order of data entry issues here. If customized section orders is too resource intensive then having the data entry orders track the hierarchy and other tables for collecting event, record event and locality, spatial details makes the most sense and would be universally logically (since it's in descending order of general usage too). Would make the  "The 'pull specific locality from verbatim' button is useful since it is currently useless if everyone is entering the specific locality before the verbatim locality now.

- [ ] make the event_verbatim_locality field including other free text fields "stretchable"/ expandable.

- [ ] Add an ```Add more``` button for the additional fields to add more attributes or identifications

- [ ] In the customization popup window, allow sorting profiles by collection

- [ ]  from Adrienne: add the function to fill began and ended date from the verbatim date

**Comment by DerekSikes** [comment]
one more: It would be great if the collection event remarks could be stretchable/expandable (if that's not already among the 'other free text fields' already slated.

**Comment by mkoo** [comment]
Yes that was my intent! I definitely need more real estate!

**Comment by dustymc** [comment]
>allow customizable ordering of sections.

A brief experiment suggests I can order the page via adding CSS to the profile (eg, in a sustainable and compatible way). I'm reluctant to add that to the edit page (eg where a minor bug could result in CMs approving things they can't see), so I wonder if we need both a rearrangement of the 'default' (including edit page) and the ability to re-order (entry page only)?

As things are set up now, place-and-time would be a section for CSS reordering. Can we agree on a static order for the contents of that? If not I could "elevate" the subsections (so eg Spatial and Locality would be their own things and could be put anywhere on the page, even very far away from each other), or possibly make them orderable within the section (which might introduce more complexity than I'm really comfortable with), or ???

>including other free text fields

They're all free-text from here! I can change any of them, but I don't know how things are used and do need a list. I'll get the ones above now.

>Add more button

That would be some huge complexity increase, I have no idea how it would work in the current setup, and I cannot see how it would be fewer clicks. Customize, change eg '2' to '3', save and use - easy-peasy?!

>sorting profiles by collection

I can add that, but it would add complexity to the forms and reduce capability - a profile could then only be used for one collection. I'm still a fan of what I suggested yesterday: Just ask the people you work with to name in some consistent manner (and don't worry about what the people you don't work with do!).

> fill began and ended date from the verbatim date

wilco

**Comment by Jegelewicz** [comment]
> As things are set up now, place-and-time would be a section for CSS reordering. Can we agree on a static order for the contents of that? 

Record event
Collecting event
Locality 

seems to be the most logical (because "The 'pull specific locality from verbatim' button is particularly useless if everyone is entering the specific locality before the verbatim locality now").

**Comment by Jegelewicz** [comment]
> And I thought we were going to get event attributes 'collection method' and 'habitat'?

https://github.com/ArctosDB/arctos/issues/6218
https://github.com/ArctosDB/arctos/issues/6217
